### ISSUE TITLE
VSCode Debugger no longer hits breakpoints

### ISSUE THREAD

**Issue by ssandler** [facebook/hhvm#8872]
**Describe the bug**
We've observed a regression in HHVM where VSCode debugging is no longer able to stop at any breakpoints. By bisecting this with hhvm nightly docker images, it appears that `hhvm/hhvm:2021.04.22` is good and `hhvm/hhvm:2021.04.23` is bad.

**Standalone code, or other way to reproduce the problem**
```
<?hh // strict

<<__EntryPoint>>
function main(): void {

	$fn = () ==> {
		var_dump('hello');
		$x = 5;
		var_dump($x);
	};

	$y = 6;
	$fn();
}

```

Set a breakpoint in every line of this file and run it in VSDebug mode (note, the setup for this especially with docker is pretty involved so hopefully you have your own ways of reproing).

We began debugging this because we observed that breakpoints stopped firing in lambda function bodies as of `2021.02.10`, and hoped that testing newer versions would yield a fix that could be backported, but unfortunately observed further regressions.

**Expected behavior**

VSCode debugger stops at breakpoints

**Actual behavior**

No breakpoints are hit

**Environment**
 - Operating system
Mac OS Catalina (with HHVM running inside docker using public docker images)
 - Installation method
hhvm/hhvm on dockerhub
 - HHVM Version
First repro is 
```
HipHop VM 4.107.0-dev (rel)
Compiler: 1619149873_805578739
Repo schema: d1ae8e21bf3419a65f12a010527485564e719d07
```

But the issue is still in place as of 
```
HipHop VM 4.123.0 (rel) (non-lowptr)
Compiler: 1629139241_178306447
Repo schema: 10a34637ad661d98ba3344717656fcc76209c2f8
```

**Comment by fredemmott** [comment]
FB T56653733

This has been a known, rare issue for quite a while at Facebook; it seems likely that this has significantly regressed this year, but we may have incorrectly considered it a report-rate change rather than a frequency change.

**Comment by ssandler** [comment]
Fwiw:
- Disabling jit does not resolve this
- `hphpd_break();` still works to stop the vscode debugger on all versions mentioned. Only VSCode's built-in breakpoints stopped working.

**Comment by fredemmott** [comment]
This has been resolved, but we're seeing a few usability issues, especially when multiple requests are in flight when a breakpoint is hit: we pause all, and vscode doesn't necessarily default to showing the correct one.

**Comment by Wilfred** [comment]
See commits 0f0ab3bf2937c3ac413a9ae9640878c86a0dad02, 197ae000930fb1ff737e737adef4e78a6c231219, 2c2d6126b5c6f4cb3d9c11fc6979801deeefcaf5, c8247a04f62b17df8de38e13c165cc25c0cd052e and f78ce7fc478aad4749ad6e3a589a0c5ed23ce432. We marked the issue as resolved in October.

There was also another fix in December in 1b9084149859848a7d3c853ee7062feccdf6ed95. I think there's nothing left to do here.
